docs/adrs/0002-handling-migration-from-sfdx-to-sf-cli-in-cumulusci.md

```diff
@@ -0,0 +1,75 @@
+
+## Decision
+
+### Considered Options
+
+1. **Run commands based on what the user has (SFDX or SF CLI):**
+
+    - Good: Gives users still using SFDX more time to switch.
+    - Good: Gives us time to fully understand the new SF CLI.
+    - Bad: It would make our code more complex and might confuse users.
+    - Bad: It divides our attention between old and new versions.
+
+2. **Stop supporting SFDX right away:**
+
+    - Good: Lets us focus on the new SF CLI and stop worrying about old code.
+    - Good: Speeds up our switch to the new CLI and helps us find and fix problems sooner.
+    - Bad: It could disrupt users' work if they're still using SFDX.
+    - Bad: There could be problems with the new SF CLI we don't know about yet.
+
+3. **Let SF CLI handle old SFDX commands:**
+    - Good: Reduces changes to our commands, making the switch smoother for users.
+    - Good: Lets us switch to the new CLI slowly, reducing risk of big issues.
+    - Bad: Delays the inevitable. If they drop deprecated commands without us noticing, it would break functionality for our users.
+    - Bad: Could delay us learning and using new features of the SF CLI.
+
+### Decision Outcome
+
+Here's how we plan to make this change:
+
+-   **Maintain Backwards Compatibility:** Our priority is to ensure that all existing processes continue to function as expected. We will not modify the existing SFDX commands in our project until SF CLI announces the version in which these commands will be deprecated.
+
+-   **Delegate maintaining backwards compatibility to the SF CLI:** We will continue using existing commands as written and rely on SF CLI to interpret these commands. We think this will make the change easier and less disruptive.
+
+-   **Warn users that are on SFDX (v7) that they need to upgrade:** We'll check the installed version of the CLI and warn users that future versions of CumulusCI will drop support.
+
+-   **Update our integration tests to include `stable-rc`:** We'll revisit PR #3558 to make sure all tasks that use the CLI are tested in our workflow, and that we're testing the weekly release candidate channel (`stable-rc`).
+
+-   **Update the Falcon web applications to depend on SF as soon as practical:** We know we need to switch to SF CLI, and we want our Falcon web apps to benefit from its latest updates and features as soon as possible. We'll also install all necessary JIT plugins in the apps, so we don't have to install them over and over during builds.
+
+-   **Update the SFDX Heroku buildpack to depend on SF as soon as practical:** Like the Falcon web applications, we want our [Heroku buildpack](https://github.com/SalesforceFoundation/simple-salesforce-dx-buildpack)to stay up-to-date with the latest SF CLI to get its performance and feature enhancements. Like the Falcon web applications, this includes installing all necessary JIT plugins for the buildpack.
```

Comment:
What criteria would you suggest for determining when it's practical to make this transition?

Reply:
The main constraint I see is the team's bandwidth:

- for testing the dependency switch from `sfdx-cli` to `@salesforce/cli`
  on the Falcon web apps
- updating the buildpack to ensure the `sf` CLI is on the `PATH` and
  that the JIT plugins are available
- triggering builds to add the latest executables to the Heroku build
  slug

I'm not seeing any technical reasons not to make the switch. Users will soon start incorporating `sf`-style commands in their automation and we'll want to support that.